src/runtime_src/core/include/experimental/xrt-next.h

```diff
@@ -276,6 +276,35 @@ XCL_DRIVER_DLLESPEC int xclOpenIPInterruptNotify(xclDeviceHandle handle, uint32_
  */
 XCL_DRIVER_DLLESPEC int xclCloseIPInterruptNotify(xclDeviceHandle handle, int fd);
 
+/**
+ * xclErrorInject() - Inject an asynchronous error record into driver
+ *
+ * @handle:     Device handle
+ * @num:        error number
+ * @driver:     driver to inject
+ * @severity:   error severity
+ * @module:     module to inject
+ * @eclass:     error class
+ * Return:      0 or appropriate error number
+ *
+ * Note: this API needs root privilege
+ */
+XCL_DRIVER_DLLESPEC
+int
+xclErrorInject(xclDeviceHandle handle, uint16_t num, uint16_t driver, uint16_t severity, uint16_t module, uint16_t eclass);
```

Comment:
Who will call this API?

Reply:
> Who will call this API?

Mainly the developers who wants to test its error handling mechanism while there is no easy way to generate error from hardware, including XRT developer, or application developers, etc.

Reply:
Ok.   This is a temporary API then?